flow360/component/simulation/models/solver_numerics.py

```diff
@@ -0,0 +1,411 @@
+"""
+Contains basic components(solvers) that composes the `volume` type models. Each volume model represents a physical phenomena that require a combination of solver features to model.
+
+E.g. 
+NavierStokes, turbulence and transition composes FluidDynamics `volume` type
+
+"""
+
+from __future__ import annotations
+
+from abc import ABCMeta
+from typing import Literal, Optional, Union
+
+import numpy as np
+import pydantic as pd
+from pydantic import NonNegativeFloat, NonNegativeInt, PositiveFloat, PositiveInt
+
+from flow360.component.simulation.framework.base_model import (
+    Conflicts,
+    Flow360BaseModel,
+)
+
+# from .time_stepping import UnsteadyTimeStepping
+
+HEAT_EQUATION_EVAL_MAX_PER_PSEUDOSTEP_UNSTEADY = 40
+HEAT_EQUATION_EVAL_FREQUENCY_STEADY = 10
+
+
+class LinearSolver(Flow360BaseModel):
+    """:class:`LinearSolver` class for setting up linear solver for heat equation
+
+
+    Parameters
+    ----------
+
+    max_iterations : PositiveInt, optional
+        Maximum number of linear solver iterations, by default 50
+
+    absolute_tolerance : PositiveFloat, optional
+        The linear solver converges when the final residual of the pseudo steps below this value. Either absolute
+        tolerance or relative tolerance can be used to determine convergence, by default 1e-10
+
+    relative_tolerance :
+        The linear solver converges when the ratio of the final residual and the initial
+        residual of the pseudo step is below this value.
+
+    validation: tolerance settings only available to HeatEquationSolver
+
+    Returns
+    -------
+    :class:`LinearSolver`
+        An instance of the component class LinearSolver.
+
+
+    Example
+    -------
+    >>> ls = LinearSolver(
+                max_iterations=50,
+                absoluteTolerance=1e-10
+        )
+    """
+
+    max_iterations: PositiveInt = pd.Field(30)
+    absolute_tolerance: Optional[PositiveFloat] = pd.Field(None)
+    relative_tolerance: Optional[PositiveFloat] = pd.Field(None)
+
+    model_config = pd.ConfigDict(
+        conflicting_fields=[Conflicts("absolute_tolerance", "relative_tolerance")]
+    )
+
+
+class GenericSolverSettings(Flow360BaseModel, metaclass=ABCMeta):
+    """:class:`GenericSolverSettings` class"""
+
+    absolute_tolerance: Optional[PositiveFloat] = pd.Field(1.0e-10)
```

Comment:
No need for Optional here?

Reply:
If the user wants to do absolute_tolerance = None it should be fine

Reply:
Then what if absolute and relative residual are both none?

Reply:
We need validation here anyways right? Absolute and relative residual cannot both be set
Actually it might be better to merge the two fields